tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1533,15 +1566,16 @@ Status StorageManager::get_fragment_info(
 
     if (meta->timestamp_range().first < timestamp_start) {
       fragment_info->expand_anterior_ndrange(
-          array_schema->domain(), non_empty_domain);
+          meta->array_schema()->domain(), non_empty_domain);
```

Comment:
`array_schema()` may return null. Later in the file, there's a comment "// Load array schema if not fetched already" that tests for null.

It may be the right thing to have `array_schema()` always load if not present. It may also be the right thing to have it load on construction. It is certainly the right thing to ensure there's not a null pointer here.

Reply:
While I certainly agree about checking and loading when missing, here on the fragment metadata the schema is always guaranteed to be set because we error on the loading of fragment metadata is the required schema is missing. This only relevant of course to the sequence I'm touching here. In general other uses of `FragmentMetadata::array_schema()` could indeed be unsafe.